### PR TITLE
Make RoR.ch ready for rails-hosting.ch

### PR DESCRIPTION
**Pull request by telmich**
Ready for merge, please change repo_url in deploy.rb to adjust to URL

### CODE REVIEW

File: config/secrets.yml [hunk 1]
```diff
@@ -0,0 +1,13 @@
+# you will have to add a file .env holding your meetup api key
```
Comment (masone) [inline config/secrets.yml]: The old implementation used Figaro for exactly this reason: all configurations were in one file. Now configurations span over `secrets.yml` and `.env`

If there is a good reason for moving to secrets instead of using Figaro, let's discuss. Otherwise, I'd revert back to Figaro.

File: .ruby-version [hunk 2]
```diff
@@ -0,0 +1 @@
+2.2.2
```
Comment (masone) [inline .ruby-version]: the ruby version is also configured in the Gemfile. please decide for one place and remove the duplication.

File: app/controllers/jobs_controller.rb [hunk 3]
```diff
@@ -0,0 +1,59 @@
+class JobsController < ApplicationController
```
Comment (masone) [inline app/controllers/jobs_controller.rb]: the whole jobs thing does not belong to this pr at all

File: app/jobs/meetup_job.rb [hunk 4]
```diff
@@ -1,7 +1,4 @@
 class MeetupJob
-  include SuckerPunch::Job
```
Comment (masone) [inline app/jobs/meetup_job.rb]: removal of SuckerPunch makes this class a non-background job. think about inheriting from ActiveJob

File: test/test_helper.rb [hunk 5]
```diff
@@ -34,5 +28,9 @@ def mock_meetup
         yield
       end
     end
+
+    def perform_job
```
Comment (masone) [inline test/test_helper.rb]: this wording is super generic. rename to perform_meetup_job or so.

File: test/models/job_test.rb [hunk 6]
```diff
@@ -0,0 +1,9 @@
+require 'test_helper'
+
+class JobTest < ActiveSupport::TestCase
+  it 'should fabricate valid object' do
+    event = Fabricate(:job)
+    assert event.valid?
+    assert event.persisted?
```
Comment (masone) [inline test/models/job_test.rb]: afaik Fabricator would raise if the event has not been persisted

File: config/schedule.rb [hunk 7]
```diff
@@ -0,0 +1,24 @@
+# Use this file to easily define all of your cron jobs.
```
Comment (masone) [inline config/schedule.rb]: i would prefer a solution with proper background jobs instead of running crons. also we had troubles with whenever at silp. can you explain the reason for moving to this solution? 

another thing that might have changed: does it still support reporting to rollbar?

File: config/initializers/secret_key_base.rb [hunk 8]
```diff
@@ -1 +0,0 @@
-Rails.application.config.secret_key_base = Figaro.env.secret_key_base
```
Comment (masone) [inline config/initializers/secret_key_base.rb]: this initializer has been completely removed. where does the value get set now? or does rails automatically read that value from secrets.yml?

File: config/initializers/rollbar.rb [hunk 9]
```diff
@@ -30,7 +32,4 @@
   #
   # You can also specify a callable, which will be called with the exception instance.
   # config.exception_level_filters.merge!('MyCriticalException' => lambda { |e| 'critical' })
-
```
Comment (masone) [inline config/initializers/rollbar.rb]: rollbar now reports inline instead of in a background worker. this is a step back. can you explain why we should roll with the inline solution?

**Comment by masone** [comment]
Thanks for the pull request! It's quite heavy as there have been changes to some things that are not related to moving to another hosting. A smaller PR would be easier to review and merge.

I have two main concerns that are also addressed in the inline comments. But it's probably easier to have the discussion here:

- Can you explain why you moved from Figaro to secrets? I feel that this is a step back as configuration now needs to be done in multiple files.
- You completely moved away from background jobs with this PR. Imho the version in this PR does not meet best practices. Is there a good reason for it?
  - there are better solutions than cronjobs and whenever
  - rollbar now reports inline instead of in the background
  - we will have to reintroduce this anyway, eg. when we want to start sending emails

File: config/schedule.rb (hunk 7 above)
Comment (geku) [inline config/schedule.rb]: I also see the need for proper background jobs for sending mails when implementing the job listing feature.

**Comment by masone** [comment]
This PR has gone stale after my review comments. In order to move on, I will take out / replay some of the changes made here. The biggest part - the design part has not been discussed yet and will be postponed.

Thanks for your help!
